### PR TITLE
moving AddDevice to react-async-saga-reducer

### PR DESCRIPTION
**Pull request by asrudra**

### CODE REVIEW

File: src/app/azureResource/sagas/setActiveAzureResourceSaga.ts [hunk 1]
```diff
@@ -3,12 +3,18 @@
...
 import { clearModelDefinitionsAction } from '../../devices/deviceContent/actions';
+import { ACTIVE_CONNECTION_STRING } from '../../constants/browserStorage';
 
 export function* setActiveAzureResourceSaga(action: Action<AzureResource>) {
+    yield call(setActiveConnectionString, action.payload.connectionString);
     yield put(clearDevicesAction());
     yield put(clearModelDefinitionsAction());
 }
+
+export const setActiveConnectionString = (value: string): void => {
```
Comment (YingXue) [inline src/app/azureResource/sagas/setActiveAzureResourceSaga.ts:18]: One thing minor: let's put some //todo here to remind us, and aslo clear the doubts if others are working on this file, as this was designed with a different purpose

File: src/app/devices/addDevice/saga.spec.ts [hunk 2]
```diff
@@ -70,15 +62,15 @@ describe('addDeviceSaga', () => {
...
+            value: call(raiseNotificationToast, {
                 text: {
                     translationKey: ResourceKeys.notifications.addDeviceOnSucceed,
                     translationOptions: {
                         deviceId
                     },
                 },
                 type: NotificationType.success
-            }))
+            });
```
Comment (YingXue) [inline src/app/devices/addDevice/saga.spec.ts:73]: unnecessary trailing semicolon

File: src/app/notifications/components/notificationToast.tsx [hunk 3]
```diff
@@ -32,12 +34,18 @@ export const CloseButton = (props: CloseButtonProps): JSX.Element => {
     );
 };
 
-const fetchComponent = (notification: Notification) => {
-    return <NotificationListEntry notification={notification} showAnnoucement={true}/>;
+const NotificationEntry = (props: { notification: Notification }): JSX.Element => {
+    const dispatch = useDispatch();
+
+    React.useEffect(() => {
+        dispatch!(addNotificationAction(props.notification));
```
Comment (YingXue) [inline src/app/notifications/components/notificationToast.tsx:41]: I am not familiar with this dispatch! syntax here, is it any different from simply dispatch?

Comment (asrudra) [reply src/app/notifications/components/notificationToast.tsx:41]: just checking dispatch is a function with !

**Pull request merged**